libraries/Microsoft.Bot.Builder.AI.Orchestrator/OrchestratorRecognizer.cs

```diff
@@ -0,0 +1,346 @@
+                ((JArray)values).Add(entity.GetValue("text", StringComparison.InvariantCulture));
+
+                // get/create $instance
+                JToken instanceRoot;
+                if (!recognizerResult.Entities.TryGetValue("$instance", StringComparison.OrdinalIgnoreCase, out instanceRoot))
+                {
+                    instanceRoot = new JObject();
+                    recognizerResult.Entities["$instance"] = instanceRoot;
+                }
+
+                // add instanceData
+                JToken instanceData;
+                if (!((JObject)instanceRoot).TryGetValue(entityResult.Type, StringComparison.OrdinalIgnoreCase, out instanceData))
+                {
+                    instanceData = new JArray();
+                    instanceRoot[entityResult.Type] = instanceData;
+                }
+
+                JObject instance = new JObject();
+                instance.Add("startIndex", entity.GetValue("start", StringComparison.InvariantCulture));
+                instance.Add("endIndex", entity.GetValue("end", StringComparison.InvariantCulture));
+                instance.Add("score", (double)1.0);
+                instance.Add("text", entity.GetValue("text", StringComparison.InvariantCulture));
+                instance.Add("type", entity.GetValue("type", StringComparison.InvariantCulture));
+                instance.Add("resolution", entity.GetValue("resolution", StringComparison.InvariantCulture));
+                ((JArray)instanceData).Add(instance);
+            }
+
+            return recognizerResult;
+        }
+
+        private void InitializeModel()
+        {
+            if (_modelPath == null)
+            {
+                throw new ArgumentNullException($"Missing `ModelPath` information.");
+            }
+
+            if (_snapshotPath == null)
+            {
+                throw new ArgumentNullException($"Missing `ShapshotPath` information.");
+            }
+
+            if (orchestrator == null)
+            {
+                var fullModelPath = Path.GetFullPath(PathUtils.NormalizePath(_modelPath));
+                Stopwatch sw = new Stopwatch();
+
+                // Create Orchestrator 
+                try
+                {
+                    sw.Start();
+                    orchestrator = new Microsoft.Orchestrator.Orchestrator(fullModelPath);
+                }
+                catch (Exception ex)
+                {
+                    sw.Stop();
+                    throw new Exception("Failed to find or load Model", ex);
+                }
+
+                sw.Stop();
+                Console.WriteLine($"Model load time:{sw.ElapsedMilliseconds}");
+            }
+
+            if (_resolver == null)
+            {
+                var fullSnapShotPath = Path.GetFullPath(PathUtils.NormalizePath(_snapshotPath));
+
+                // Load the snapshot
+                string content = File.ReadAllText(fullSnapShotPath);
```

Comment:
This seems a little awkward - going via a single big string. Maybe the file should be base64 encoded, maybe we could stream directly into the binary. Basically, there are faster ways to do this type of stuff in modern C# (ReadOnlySpans) actually I thought we were going to use that on the interface to the underlying.

No need to block on this check in - but perhaps open an issue to look at performance here.

Reply:
Sure. Open to optimization for preview 2 (R11)